client/src/templates/Challenges/redux/execute-challenge-saga.js

```diff
@@ -107,9 +107,22 @@ export function* executeChallengeSaga({ payload }) {
     yield put(initLogs());
     yield put(initConsole(i18next.t('learn.running-tests')));
     // reset tests to initial state
-    const tests = (yield select(challengeTestsSelector)).map(
+    let tests = (yield select(challengeTestsSelector)).map(
       ({ text, testString }) => ({ text, testString })
     );
+
+    const generateUniqueTestName = (testText, index) => {
+      const testName = `Test ${index + 1}`;
+      return testText.includes(testName)
+        ? testText
+        : `${testName}: ${testText}`;
+    };
```

Comment:
```suggestion
    const generateUniqueTestName = (testText, index) => /^\d+\./.test(testText) ? testText : `${index}. ${testText}`
```

Rather than prefixing each of them with `Test`, I think only enumerating them would be enough. This change should check if it starts with an ordinal number, and append it if not.